src/app/core/entity/schema-datatypes/datatype-date-only.spec.ts

```diff
@@ -35,4 +35,28 @@ describe("Schema data type:Date", () => {
     expect(date.getMonth()).toBe(3);
     expect(date.getDate()).toBe(1);
   });
+
+  fit("should migrate ISO date strings (created with timezone offsets) to a date-only", () => {
+    const testCases = [
+      { input: "2022-01-01", expected: "2022-01-01" },
+      // clean midnight dates are transformed to the inferred day
+      { input: "2023-01-20T00:00:00.000Z", expected: "2023-01-20" },
+      { input: "2022-12-31T22:00:00.000Z", expected: "2023-01-01" },
+      { input: "2022-12-31T19:30:00.000Z", expected: "2023-01-01" },
+      { input: "2023-01-01T05:00:00.000Z", expected: "2023-01-01" },
+      // exact dates do not allow to safely guess offset
+      // TODO: we could store these as "createdAt" of the entity (once that is implemented ...)
+      { input: "2023-01-06T10:03:35.726Z", expected: "2023-01-06" },
+      { input: "2023-01-06T23:59:39.726Z", expected: "2023-01-06" }, // TODO: what to do here? these are actually not deterministic ...
```

Comment:
I would say the moment we have an accurate time (minutes, seconds, milliseconds not 0) we should just use the date.
If minutes, seconds and milliseconds are 0 we know it has to be a flat date and we can use the closest one.